### ISSUE TITLE
Lockup/freeze when viewing diff for submodule change

### ISSUE THREAD

**Issue by mdonatas** [gitextensions/gitextensions#9120]
## Current behaviour

GitExtensions `master` (`7d25651ea1da86efe5d0c7119092868d85a7d754`) branch build freezes up when opening diff for commit `31650842179659699f38371861629ae403f88795` It has just 1 submodule changed.

I didn't have time to bisect or investigate this yet.

## Expected behaviour

No freeze.

## Screenshots

Threads during the freeze
![image](https://user-images.githubusercontent.com/483659/116073250-d264e700-a698-11eb-86e4-93df9613dbfe.png)

There is a stream of `TaskCanceledException` in `Output`
```
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
Exception thrown: 'System.Threading.Tasks.TaskCanceledException' in mscorlib.dll
```

## Did this work in previous version of GitExtensions

It works OK in 3.5 branch

## Environment

- Git Extensions 33.33.33
- Build 7d25651ea1da86efe5d0c7119092868d85a7d754
- Git 2.31.0.windows.1
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4341.0
- DPI 192dpi (200% scaling)

**Comment by drewnoakes** [comment]
Possible dupe of #8914. Fixed in #8923.

**Comment by mdonatas** [comment]
Whoops, I went a bit back to see if this was reported already but obviously didn't go back long enough. Thanks!

**Comment by gerhardol** [comment]
With #8923 this is an improvement to #8881 so I wanted to fix the test issues blocking #8923 but there has been some other issues to handle. Will create a short term fix.
